sdks/java/extensions/python/src/main/java/org/apache/beam/sdk/extensions/python/PythonService.java

```diff
@@ -96,6 +95,18 @@ public AutoCloseable start() throws IOException, InterruptedException {
     return p::destroy;
   }
 
+  private String whichPython() {
+    for (String executable : ImmutableList.of("python3", "python")) {
+      try {
+        new ProcessBuilder(executable, "--version").start().waitFor();
+        return executable;
+      } catch (IOException | InterruptedException exn) {
+        // Ignore.
```

Comment:
Will ignore result in a default Python binary being picked ?

Reply:
It will move on to the next item in the list above, trying to find some python in the path. In this case, first we're trying python3, and if that fails we try plain old python, and if that fails we give up (though we could add other items in this list as needed).